src/library_tv/library_tv.c

```diff
@@ -0,0 +1,284 @@
+#include <nitro.h>
+#include <string.h>
+
+#include "message.h"
+#include "struct_decls/struct_02018340_decl.h"
+
+#include "struct_defs/struct_0205AA50.h"
+#include "struct_defs/struct_02099F80.h"
+#include "overlay084/struct_ov84_0223BA5C.h"
+#include "overlay097/struct_ov97_0222DB78.h"
+
+#include "unk_02002B7C.h"
+#include "overlay_manager.h"
+#include "unk_02006E3C.h"
+#include "message.h"
+#include "unk_0200F174.h"
+#include "unk_02017728.h"
+#include "heap.h"
+#include "unk_02018340.h"
+#include "unk_0201D670.h"
+#include "gx_layers.h"
+
+#include "library_tv/library_tv.h"
+#include "constants/narc.h"
+
+typedef struct LibraryTV {
+    int heapID;
+    BGL * bgl;
+    MessageLoader * msgLdr;
+    int tvState;
+    Window tvWindow;
+    int scanLinePos;
+    int waitTiming;
+} LibraryTV;
+
+extern void sub_02000EC4(FSOverlayID param0, const OverlayManagerTemplate * param1);
+static void LibraryTV_VBlank(void * data);
+static void LibraryTV_SetVramBank(LibraryTV * ltv);
+static void LibraryTV_ReleaseVramBank(LibraryTV * ltv);
+static void LibraryTV_SetMsgLdr(LibraryTV * ltv);
+static void LibraryTV_ReleaseMsgLdr(LibraryTV * ltv);
+static void LibraryTV_UpdateScanLines(LibraryTV * ltv);
+
+BOOL LibraryTV_Init (OverlayManager * ovy, int * state)
+{
+    int heapID = HEAP_ID_LIBRARY_TV;
+
+    Heap_Create(HEAP_ID_APPLICATION, heapID, HEAP_SIZE_LIBRARY_TV);
+
+    LibraryTV * ltv = OverlayManager_NewData(ovy, sizeof(LibraryTV), heapID);
+    memset(ltv, 0, sizeof(LibraryTV));
+
+    ltv->heapID = heapID;
+    ltv->waitTiming = 0;
+
+    return TRUE;
+}
+
+enum {
+    STATE_INIT = 0,
+    STATE_FALLTHROUGH = 1,
+    STATE_FADE_START = 2,
+    STATE_FADE_WAIT = 3,
+    STATE_MAIN = 4,
+    STATE_EXIT = 5
+};
+
+#define LIBRARY_TV_DURATION 90 + 150
+
+BOOL LibraryTV_Main (OverlayManager * ovy, int * state)
+{
+    LibraryTV * ltv = OverlayManager_Data(ovy);
+    BOOL result = FALSE;
+
+    switch (*state) {
+    case STATE_INIT:
+        sub_0200F344(0, 0x0);
+        sub_0200F344(1, 0x0);
+
+        SetMainCallback(NULL, NULL);
+        SetHBlankCallback(NULL, NULL);
+
+        GXLayers_DisableEngineALayers();
+        GXLayers_DisableEngineBLayers();
+
+        GX_SetVisiblePlane(0);
+        GXS_SetVisiblePlane(0);
+
+        SetAutorepeat(4, 8);
+
+        LibraryTV_SetVramBank(ltv);
+        LibraryTV_SetMsgLdr(ltv);
+
+        SetMainCallback(LibraryTV_VBlank, (void *)ltv);
+        GXLayers_TurnBothDispOn();
+
+        ltv->waitTiming = 0;
+        *state = STATE_FALLTHROUGH;
+        break;
+    case STATE_FALLTHROUGH:
+        if (ltv->waitTiming) {
+            ltv->waitTiming--;
+        } else {
```

Comment:
**question:** Is this block logically unreachable? If so, leave a comment regarding it. This also applies to the same block in `STATE_FADE_START`.